scripting/mastersConcordance.js

```diff
@@ -0,0 +1,159 @@
+// Masters Concordance Calculations
+
+// Input Paramaters:
+// (reference, query, mode, amelogenin = true)
+// Modes:
+// 0 = non-empty markers, 1 = query markers, 2 = reference markers
+
+// Output:
+// float from 0 to 100 (%), representing concordance
+
+// count number of shared alleles between one query locus and one reference locus
+function testLoci(testQ, testR) {
+  let shared = 0;
+
+  /*
+  // transform alleles to number (int/float) values OR, for amelogenin, to uppercase
+  testQ = testQ.map(x => (isNaN(Number(x)) ? x.toUpperCase() : Number(x)));
+  testR = testR.map(x => (isNaN(Number(x)) ? x.toUpperCase() : Number(x)));
+
+  // filter out all non-unique values within each locus
+  // in case the input does not conform with the CLASTR format (no repeated values)
+  let distinct = (value, index, self) => self.indexOf(value) === index;
+  testQ.filter(distinct);
+  testR.filter(distinct);
+  */
+
+  // test if each allele from the query locus is shared in the reference locus
+  for (let allele of testQ) {
+    if (testR.includes(allele)) {
+      shared++; // count shared alleles
+    }
+  }
+
+  return shared;
+}
+
+// concordance: masters vs query
+function mastersQuery(reference, query, mode, amelogenin = true) {
+  // shared alleles ÷ query alleles
+  let shared = 0;
+  let queryAlleles = 0;
+
+  if (mode == 0 || mode == 2) {
```

Comment:
The OR operator is used because only one mode can be selected at a time. The reason why it's for both reference and non-empty markers is because the exact same code can be used for both of them.